### PR TITLE
Test: Re-write unit tests

### PR DESCRIPTION
**Pull request by SukkaW**
@kdy1 I have rewritten the unit test to make it works.

### CODE REVIEW

File: __tests__/browserify.js [hunk 1]
```diff
@@ -3,18 +3,20 @@ const path = require("path");
 const vm = require("vm");
 
 describe("browserify", function() {
-  it("@swc/register may be used without breaking browserify", function(done) {
+  it("@swc/register may be used without breaking browserify", () => {
```
Comment (SukkaW) [inline __tests__/browserify.js:6]: Rewrite callback-based async test case into Promise.

File: src/node.ts [hunk 2]
```diff
@@ -61,7 +61,7 @@ function compile(code: string | any, filename: string) {
   delete opts.ignore;
   const output: swc.Output = swc.transformSync(code, {
     ...opts,
-    sourceMaps: opts.sourceMaps === undefined ? "inline" : opts.sourceMaps
+    sourceMaps: opts.sourceMaps === undefined ? true : opts.sourceMaps
```
Comment (SukkaW) [inline src/node.ts:64]: Babel uses `sourceMaps: 'both'` here, which is not supported by swc. So I change it to `true` to match the behavior of Babel.

File: src/node.ts [hunk 3]
```diff
@@ -89,15 +89,13 @@ function compileHook(code: string, filename: string) {
 
 function hookExtensions(exts: readonly string[]) {
   if (piratesRevert) piratesRevert();
-  piratesRevert = addHook(compileHook, { exts: exts as string[], ignoreNodeModules: true });
+  piratesRevert = addHook(compileHook, { exts: exts as string[], ignoreNodeModules: false });
```
Comment (SukkaW) [inline src/node.ts:92]: Here I set `ignoreNodeModules` to `false` in order to match the behavior of Babel. The ignoring `node_modules` behavior will be done by #13 instead.

File: src/node.ts [hunk 4]
```diff
@@ -89,15 +89,13 @@ function compileHook(code: string, filename: string) {
...
 }
 
 export function revert() {
   if (piratesRevert) piratesRevert();
 }
 
-register();
```
Comment (SukkaW) [inline src/node.ts:99]: `@babel/register` doesn't have it.

**Review by kdy1 — APPROVED** [review]
LGTM. Thank you and sorry for the delay. I had too many tasks to do.

**Pull request merged**